### PR TITLE
Org Ledger: Filter transactions by date

### PR DESCRIPTION
**Pull request by manuthecoder**
## Summary of the problem
Add date range functionality to filters

## Describe your changes
#4958

### CODE REVIEW

**Comment by manu-gurudath** [comment]
yikes I was editing on the wrong branch so the deployment might fail

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -75,6 +75,8 @@ def show
     @user = User.find(params[:user]) if params[:user]
 
     @type = params[:type]
+    @startDate = params[:start]
+    @endDate = params[:end]
```
Comment (sampoder) [inline app/controllers/events_controller.rb]:
```suggestion
    @start_date = params[:start]
    @end_date = params[:end]
```

I know this pedantic but convention is snake casing

File: app/javascript/controllers/filter_controller.js [hunk 2]
```diff
@@ -40,6 +44,24 @@ export default class extends Controller {
...
+      this.selectUsers()
+      this.selectDates()
+      if (this.hasTagsTarget) {
+        this.selectTags()
+      }
+    }
+  }
+
+  selectDates(mouseEvent) {
+    console.log('triggered')
```
Comment (sampoder) [inline app/javascript/controllers/filter_controller.js]:
```suggestion
```

Let's remove all the `console.log()`s in here :D

**Review by sampoder — COMMENTED** [review]
Couple of quick changes. Do we want to support only setting a start date or only setting a end date? 

An alternative would be setting the default values of those fields to the date of the first transaction (start) and today's date (end).

**Review by sampoder — APPROVED** [review]
Looks good to me!

File: app/views/events/_filter.html.erb [hunk 3]
```diff
@@ -62,6 +73,13 @@
           style="cursor: pointer;">
           Types
         </span>
+        <span
+          data-filter-target="datesLabel"
+          data-filter-menu-label-selected="<%= @dates ? "true" : "false" %>"
```
Comment (garyhtou) [inline app/views/events/_filter.html.erb]:
```suggestion
          data-filter-menu-label-selected="<%= @start_date || @end_date ? "true" : "false" %>"
```

File: app/views/events/_filter.html.erb [hunk 4]
```diff
@@ -104,7 +122,19 @@
         <% end %>
       </div>
 
-      <% if @tag || @user || @type %>
+      <div data-filter-target="dates" data-filter-menu-selected="<%= @dates ? "true" : "false" %>">
```
Comment (garyhtou) [inline app/views/events/_filter.html.erb]:
```suggestion
      <div data-filter-target="dates" data-filter-menu-selected="<%= @start_date || @end_date ? "true" : "false" %>">
```

File: app/views/events/_filter.html.erb [hunk 5]
```diff
@@ -104,7 +122,19 @@
...
+          <%= form.date_field :start, class: "border mb1", value: @start_date %>
+          <p class="bold mb0 mt0">Transactions after...</p>
+          <%= form.date_field :end, class: "border mb1", value: @end_date %>
+          <div class="flex justify-start">
+            <%= form.submit "Filter Transactions" %>
+          </div>
+        <% end %>
+      </div>
+
+      <% if @tag || @user || @type || @start_date %>
```
Comment (garyhtou) [inline app/views/events/_filter.html.erb]:
```suggestion
      <% if @tag || @user || @type || @start_date || @end_date %>
```

File: app/controllers/events_controller.rb [hunk 6]
```diff
@@ -90,6 +92,15 @@ def show
       @pending_transactions = @pending_transactions.select { |x| x.stripe_cardholder && x.stripe_cardholder.user.id == @user.id }
     end
 
+    if @start_date || @end_date
+      @all_transactions = @all_transactions.select { |t|
+        (!@start_date || t.date >= @start_date.to_datetime) && (!@end_date || t.date <= @end_date.to_datetime)
+      }
+      @pending_transactions = @pending_transactions.select { |x|
+        (!@start_date || x.date >= @start_date.to_datetime) && (!@end_date || x.date <= @end_date.to_datetime)
+      }
```
Comment (garyhtou) [inline app/controllers/events_controller.rb]:
```suggestion
    if @start_date || @end_date
      in_range = ->(t) { (!@start_date || t.date >= @start_date.to_datetime) && (!@end_date || t.date <= @end_date.to_datetime) }

      @all_transactions = @all_transactions.select(&in_range)
      @pending_transactions = @pending_transactions.select(&in_range)
```

**Review by garyhtou — APPROVED** [review]

**Pull request merged**